openml/tasks/functions.py

```diff
@@ -441,3 +442,68 @@ def _create_task_from_xml(xml):
         raise NotImplementedError('Task type %s not supported.' %
                                   common_kwargs['task_type'])
     return cls(**common_kwargs)
+
+
+def create_task(
+        task_type_id: int,
+        dataset_id: int,
+        target_name: str,
```

Comment:
I'm surprised this works, shouldn't it `target_name: Optional[str]` as the target is optional for clustering tasks?

Reply:
Nicely spotted, it should be optional. This raises a nice problem I guess, the unit test runs the `create_task` function with `target_name` available for all task types. 
I can modify it, to run clustering tasks without a target_name. With a `target_name` it covers the case with extra information, however in this case it backfired.
You think we should check for both? 
Or you think now that this is covered we can leave it like that.

Reply:
> You think we should check for both?

Yes, it would be good to test both.